src/panelSocket.ts

```diff
@@ -113,4 +117,26 @@ export class PanelSocket extends EventEmitter {
 
         this.isConnected = false;
     }
+
+    private registerForCDPEvents() {
+        // register for custom events from jsdebug:
+        const registrationMessage = {
+            method: 'JsDebug.subscribe',
+            params: {
+                events: [
+                'Runtime.*',
+                'DOM.*',
+                'CSS.*',
+                'DOMDebugger.*',
+                'Network.*',
+                'Page.*',
+                'Target.*',
+                'Overlay.*',
+                ],
+            },
+        };
+        if (this.socket) {
+            this.socket.send(JSON.stringify(registrationMessage));
+        }
```

Comment:
I'm a little confused by what we are doing in here. AFAIK this module represents the "real-socket" we use to establish the connection with the browser, so not sure what is the role of JsDebug.subscribe, would you mind adding some details on the expected workflow in the description?

Reply:
In the case of sharing CDP connection, this socket is provided by jsDebug, not the browser, and everything is proxied. By default, using the shared JsDebug proxy does not send any events unless specifically subscribed to. More info is available here: https://github.com/connor4312/cdp-proxy-requestor. I'll add docs here as well before checking in

Reply:
thanks for details, I think I get it what it is happening now. The approach seems the correct one to me but I think we need two classes for this:
- PanelSocket: the usual socket we know.
- DebugProxyPanelSocket: the specific implementation, (could extend PanelSocket for code reusing code)

That will help if the behavior fork in the future

Reply:
Good call. I've created a new debug proxy socket class. Helped get rid of a bunch of passing booleans into contructors